src/proxy/outbound.rs

```diff
@@ -381,14 +382,48 @@ impl OutboundConnection {
             });
         }
 
-        let us = us.unwrap();
+        let mut mutable_us = us.unwrap();
+        let workload_ip = self
+            .pi
+            .state
+            .load_balance(
+                &mutable_us.workload,
+                &source_workload,
+                &self.pi.cfg.dns_resolver_config,
+                &self.pi.cfg.dns_resolver_opts,
+                self.pi.metrics.clone(),
+            )
+            .await?;
+        let us = match set_gateway_address(&mut mutable_us, workload_ip, self.pi.hbone_port) {
```

Comment:
Do we need to call this here, or does the call within fetch_waypoint do enough for us?

Reply:
this originally kept the original behavior (in case of waypoint, we called `set_gateway_address` twice. that said, we only really need to do it once. so i updated it in https://github.com/istio/ztunnel/pull/612/commits/630671d02e0cab2ed676da33becb613f38be4737